lib/services.rb

```diff
@@ -23,3 +11,31 @@ def self.publishing_api
     )
   end
 end
+
+# First attempt at a retry-thing. Keeping it inside this repo for ease of
+# testing. Once it's good we'll move this up to GdsApi adapters.
+module GdsApi
+  def self.with_retries(maximum_number_of_retries = 4)
+    attempts = 0
+    maximum_number_of_attempts = maximum_number_of_retries + 1
+
+    begin
+      attempts += 1
+      yield
+    rescue Timeout::Error, GdsApi::TimedOutException => e
+      raise e if attempts >= maximum_number_of_attempts
+      sleep sleep_time_after_attempt(attempts)
```

Comment:
@tijmenb glad to see this - separating effects from calculations is a good habit to get into.